components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java

```diff
@@ -529,6 +529,146 @@ public AccessTokenDO retrieveLatestAccessToken(String consumerKey, Authenticated
         }
     }
 
+    /**
+     * This method is introduced to fix
+     * IDENTITY-5827: Generating refresh tokens within small time period throws errors
+     */
+    public List<AccessTokenDO> retrieveLatestAccessTokens(String consumerKey, AuthenticatedUser authzUser,
+                                                          String userStoreDomain, String scope,
+                                                          boolean includeExpiredTokens, int limit)
+            throws IdentityOAuth2Exception {
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection();
+        boolean isUsernameCaseSensitive = IdentityUtil.isUserStoreInUsernameCaseSensitive(authzUser.toString());
+        String tenantDomain = authzUser.getTenantDomain();
+        int tenantId = OAuth2Util.getTenantId(tenantDomain);
+        String tenantAwareUsernameWithNoUserDomain = authzUser.getUserName();
+        String userDomain = getSanitizedUserStoreDomain(authzUser.getUserStoreDomain());
+        userStoreDomain = getSanitizedUserStoreDomain(userStoreDomain);
```

Comment:
Good to have a comment what is the difference of these two user domains. It can be seen one is passed and other is taken from authzUser. But there meaningful difference in usage is not very clear.